mesonbuild/dependencies.py

```diff
@@ -598,12 +598,18 @@ def __init__(self, environment, kwargs):
             self.boost_root = None
         if self.boost_root is None:
             if self.want_cross:
-                raise DependencyException('BOOST_ROOT is needed while cross-compiling')
+                if 'BOOST_INCLUDEDIR' in os.environ:
+                    self.incdir = os.environ['BOOST_INCLUDEDIR']
+                else:
+                    raise DependencyException('BOOST_ROOT or BOOST_INCLUDEDIR is needed while cross-compiling')
             if mesonlib.is_windows():
                 self.boost_root = self.detect_win_root()
                 self.incdir = self.boost_root
             else:
-                self.incdir = '/usr/include'
+                try:
```

Comment:
Why not use `if 'BOOST_INCLUDEDIR' in os.environ:` here too? Just for consistency.